File: week01/day07/frontend/src/types/session.ts
export interface Session {
  id: number;
  patient_id: number;
  worker_id: number;
  status: string;
  summary?: string;
  created_at: string;
  patient_name?: string;
}

export interface ChatMessage {
  id: number;
  session_id: number;
  role: 'user' | 'assistant';
  content: string;
  created_at: string;
}

export interface WorkerPatient {
  id: number;
  name: string;
  age: number;
  gender: string;
  insurance_type: string;
  guardian_info?: string;
  disease_info?: string;
  care_requirements?: string;
  personality?: string;
  info_completeness: number;
  has_ongoing_session: boolean;
}

export interface ExtractResult {
  guardian_info?: string;
  disease_info?: string;
  care_requirements?: string;
  personality?: string;
}

export interface SessionResponse {
  session: Session;
  messages: ChatMessage[];
}

export interface AddMessageResponse {
  user_message: ChatMessage;
  ai_message: ChatMessage;
}

File: week01/day07/backend/schemas/session.py
from pydantic import BaseModel, Field


class SessionCreate(BaseModel):
    patient_id: int = Field(..., ge=1)


class MessageAdd(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)


class SessionOut(BaseModel):
    id: int
    patient_id: int
    worker_id: int
    status: str
    summary: str | None = None
    created_at: str
    patient_name: str | None = None

    model_config = {"from_attributes": True}


class MessageOut(BaseModel):
    id: int
    session_id: int
    role: str
    content: str
    created_at: str

    model_config = {"from_attributes": True}


class ExtractResult(BaseModel):
    guardian_info: str | None = None
    disease_info: str | None = None
    care_requirements: str | None = None
    personality: str | None = None


class ConfirmSubmit(BaseModel):
    guardian_info: str | None = None
    disease_info: str | None = None
    care_requirements: str | None = None
    personality: str | None = None
